document-portal/xdg-permission-store.c

```diff
@@ -506,6 +506,48 @@ handle_set_value (XdgPermissionStore     *object,
   return TRUE;
 }
 
+static gboolean
+handle_get_tables (XdgPermissionStore     *object,
+                       GDBusMethodInvocation  *invocation)
+{
+  g_autoptr(GVariantBuilder) builder = NULL;
+  g_autofree char *path = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDir) dir = NULL;
+  g_autofree const gchar *filename = NULL;
+
+  builder = g_variant_builder_new (G_VARIANT_TYPE ("as"));
+
+  path = g_build_filename (g_get_user_data_dir (), "flatpak/db", NULL);
+
+  if (!g_file_test (path, G_FILE_TEST_IS_DIR))
+   {
+     g_dbus_method_invocation_return_value (invocation, g_variant_new ("(as)", builder));
+     return TRUE;
+   }
+
+  dir = g_dir_open(path, 0, &error);
+
+  if (error != NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_FAILED,
+                                             "Unable to list tables: %s", error->message);
+      return TRUE;
+    }
+
+  while ((filename = g_dir_read_name(dir)))
+    {
+      g_variant_builder_add (builder, "s", filename);
```

Comment:
This means you will add everything in that directory as a table. I think you this should only happen if the file is actually a table.

Reply:
How do I know if it is a table?